packages/mobile-native-mcp-server/src/tools/plan/sfmobile-native-project-generation/tool.ts

```diff
@@ -67,36 +64,27 @@ export class SFMobileNativeProjectGenerationTool extends AbstractNativeProjectMa
   ): string {
     const platformLower = input.platform.toLowerCase();
 
+    // Build template properties flags if they exist
+    let templatePropertiesFlags = '';
+    if (input.templateProperties && Object.keys(input.templateProperties).length > 0) {
+      const propertyFlags = Object.entries(input.templateProperties)
+        .map(([key, value]) => `--template-property-${key}="${value}"`)
+        .join(' ');
+      templatePropertiesFlags = ` ${propertyFlags}`;
+    }
+
     return dedent`
       ## Step ${stepNumber}: Execute Platform-Specific CLI Command
 
       Generate the project using the Salesforce Mobile SDK CLI:
 
       \`\`\`bash
-      sf mobilesdk ${platformLower} createwithtemplate --templatesource="${MOBILE_SDK_TEMPLATES_PATH}" --template="${input.selectedTemplate}" --appname="${input.projectName}" --packagename="${input.packageName}" --organization="${input.organization} --consumerkey="${input.connectedAppClientId}" --callbackurl="${input.connectedAppCallbackUri}" --loginserver="${input.loginHost}"
+      sf mobilesdk ${platformLower} createwithtemplate --templatesource="${MOBILE_SDK_TEMPLATES_PATH}" --template="${input.selectedTemplate}" --appname="${input.projectName}" --packagename="${input.packageName}" --organization="${input.organization} --consumerkey="${input.connectedAppClientId}" --callbackurl="${input.connectedAppCallbackUri}" --loginserver="${input.loginHost} ${templatePropertiesFlags}"
       \`\`\`
 
       **Expected Outcome**: A new ${input.platform} project directory named "${input.projectName}" will be created with the template structure. The output of the command will indicate the location of the bootconfig.plist file, take note of this for oauth configuration!
```

Comment:
bootconfig.plist file thing is iOS specific,  need something for android? or the template.js from the template should do the token to the command line property value replacement.